### ISSUE TITLE
connection problems after running 'generateGravityWildcards.sh'

### ISSUE THREAD

**Issue by EPiC-APOC** [mmotti/pihole-gravity-wildcards#1]
what I did and what happened:
```
pi@RPiHole:~ $ sudo bash
root@RPiHole:/home/pi# wget -qO /usr/local/bin/generateGravityWildcards.sh https://raw.githubusercontent.com/mmotti/pihole-gravity-wildcards/master/generateGravityWildcards.sh
root@RPiHole:/home/pi# chmod +x /usr/local/bin/generateGravityWildcards.sh
root@RPiHole:/home/pi# exit
exit
pi@RPiHole:~ $ generateGravityWildcards.sh
[i] Updating gravity
[i] Fetching domains from gravity.list
[i] Identifying domains with >= 20 subdomains
[i] 933 domains found
[i] Fetching domains from whitelist.txt
[i] Checking for whitelist conflicts
[i] 929 domains remain after conflict resolution
[i] Determining blocking mode
[i] Outputting domains to: /etc/dnsmasq.d/mmotti_generated_wildcards.conf
[i] Restarting Pi-hole service
pi@RPiHole:~ $ pihole restartdns reload
  [✗] pihole-FTL: no process found
pi@RPiHole:~ $ pihole restartdns
  [✓] Starting DNS service
pi@RPiHole:~ $ pihole restartdns reload
  [✗] pihole-FTL: no process found
pi@RPiHole:~ $ sudo service pihole-FTL restart
pi@RPiHole:~ $ pihole restartdns
  [✓] Restarting DNS service
pi@RPiHole:~ $ pihole restartdns reload
  [✓] Reloading DNS service
pi@RPiHole:~ $ 
```
Using up2date Raspbian Stretch Lite on a Raspberry Pi 2B with latest Pi-Hole and also running DNScrypt 2.0

I loose connection right after running `generateGravityWildcards.sh`, as you can see as mentioned above.
Got connection back right after `sudo service pihole-FTL restart` & `pihole restartdns` & `pihole restartdns reload` now stated `Reloading DNS service` instead of `no process found` (as you can see).

**Comment by mmotti** [comment]
Thanks for raising this issue.

It looks as though the script has run through to successfully generate the 929 wildcards.

When exactly is the connection lost? Is it immediately after the script completes or some time after?

Also - Did you run those commands immediately after the script finished processing?

There is a `sudo service pihole-Ftl restart` at the end of the script which should reload things properly.

If you can, it may be worth uploading your `mmotti_generated_wildcards.conf` and also looking through the FTL log file to see if anything obvious is in there.

**Comment by EPiC-APOC** [comment]
Connection loss happened pretty immediatly after script finish (sometimes earlier it happened with a small delay).

This commands I used as soon as I saw my connection was lost, pretty quick this time 😉 and I also used `sudo service pihole-Ftl restart` (as you can see). 

I'll upload you the 2 requested files. 

With kind regards

**Comment by mmotti** [comment]
OK - it's not clear at the moment what would cause this.

The confusing part is that the dns resolution may be having issues after running the script (which runs a service restart at the end), yet after an additional service restart is apparently OK... even though the conf file we added is still there.

All the script does is generate a conf file and place it in a directory. If it were an issue with the script / conf file, it should fail also when you try a manual service restart.

Hooefully the logs will clear things up :-)

**Comment by EPiC-APOC** [comment]
1st few times I also had localhost in the .conf, now this does not seem to happen anymore. Tested it now a few times but no problems with my connection yet, see what happens after a little longer. I also do not get the localhost entry back, I didn't change settings or whatsoever for gravity-wildcards.

(I did save some logs for bug hunting)

If this keeps going allright I'll try to re-enable the cronjob. I'll keep you posted. At the moment I can't seem to reproduce it... Yet... 😉
